src/pykeen/nn/node_piece/utils.py

```diff
@@ -19,10 +19,24 @@
 def random_sample_no_replacement(
     pool: Mapping[int, Collection[int]],
     num_tokens: int,
+    num_entities: int,
```

Comment:
Perhaps make this optional and calculate it as the length of the pool of not given.

Reply:
good catch!